lib/screens/home/accounts/accounts_screen.dart

```diff
@@ -25,31 +26,34 @@ class AccountsScreen extends StatefulWidget {
 class AccountsScreenState extends State<AccountsScreen> {
   final _listKey = GlobalKey<AnimatedListState>();
   final _subscriptions = CompositeSubscription();
-  final _accountsBloc = AccountsBloc();
+  final _providerSubscriptions = CompositeSubscription();
   final _accountService = get<AccountService>();
+  late final AccountsBloc _bloc;
 
   @override
   void dispose() {
     _subscriptions.dispose();
-
     super.dispose();
-
-    get.unregister<AccountsBloc>();
   }
 
   @override
   void initState() {
     super.initState();
 
-    get.registerSingleton(_accountsBloc);
-
-    _accountsBloc.load();
-
     _accountService.events.removed.listen(_onRemoved).addTo(_subscriptions);
-    _accountsBloc.insert.listen(_onInsert).addTo(_subscriptions);
-    _accountsBloc.loading
-        .listen((e) => _onLoading(context, e))
-        .addTo(_subscriptions);
+  }
+
+  @override
+  void didChangeDependencies() {
+    _bloc = Provider.of<AccountsBloc>(context);
+    _providerSubscriptions.cancel().whenComplete(() {
```

Comment:
Since `cancel` calls `dispose` in its implementation, `_providerSubscriptions` will not be able to be used a second time. We could call the `clear` method, but waiting for cancellation in order to listen may also cause this widget to miss events.

One solution would be:
```
StreamSubscription _providerSubscription;
...
_providerSubscription?.cancel();
_providerSubscription = _bloc.insert.listen(_onInsert);
```
Or if we want a composite subscription:
```
CompositeSubscription _providerSubscription;
...
_providerSubscription?.cancel();
final providerSubscription = CompositeSubscription();
_bloc.insert.listen(_onInsert).addTo(providerSubscription);
_providerSubscription = providerSubscription;
```